### PR TITLE
[Ragile]Adapt kernel 5.10 for broadcom on RA-B6510-48V8C

### PR DESCRIPTION
**Pull request by pettershao-ragilenetworks**
#### Why I did it
Adapt kernel 5.10 for broadcom on RA-B6510-48V8C

#### How I did it
Adapt kernel 5.10 for broadcom on RA-B6510-48V8C

#### How to verify it
show version
docker ps -a
show platform syseeprom
show platform fan
show platform psustatus
show platform temperature
show platform firmware status
show platform ssdhealth
show interface status
show interface transceiver presence
show interface transceiver eeprom
show environment
sensors
sudo show system-health detail

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] master

### CODE REVIEW

**Comment by stephengao-ragilenetworks** [comment]
@lguohan  , please take a time to review it, thanks a lot! It's been three weeks since this PR opened, RA-B6510-48V8C is a very important platform for us to make verification, especially our test team is working on a new contribution of a sonic testbed plugin and verifying on this platform. So please take a time! Cheers. @zhangyanzhao ，please help to push this progress, very grateful.

**Comment by philo-micas** [comment]
[6510 Software & hardware interface test report1.docx](https://github.com/sonic-net/sonic-buildimage/files/11854121/6510.Software.hardware.interface.test.report1.docx)

**Comment by pettershao-ragilenetworks** [comment]
[Software & hardware interface test report.docx](https://github.com/sonic-net/sonic-buildimage/files/11854144/Software.hardware.interface.test.report.docx)
@lguohan help check test report, thanks!

**Comment by pettershao-ragilenetworks** [comment]
[report_6510.zip](https://github.com/sonic-net/sonic-buildimage/files/11953086/report_6510.zip)
@lguohan  @yxieca  newest pytest report with version  include platform.json, help check and forward, thanks!

**Review by prgeor — COMMENTED** [review]
@pettershao-ragilenetworks why include firmware binary .bin files as part of sonic image?

**Comment by pettershao-ragilenetworks** [comment]
> @pettershao-ragilenetworks why include firmware binary .bin files as part of sonic image?

thanks for review, these firmware bins are use to adapt two commands: `show platform firmware updates`  and   `fwutil update all fw` since we can't modify fwutil file, and other vendors seem that have't adapt these two commands at all, and their platform_components.json files are all empty, and we can remove them once community fix this issue, and please check our test report for these functions:
[RA-B6510-48V8C_check_log.log](https://github.com/sonic-net/sonic-buildimage/files/11973875/RA-B6510-48V8C_check_log.log)

**Comment by Stephengzh-Ragilenetworks** [comment]
@prgeor Hi Prince, can you take the time to check the answers from Petter? We urgently need to merge this PR this week, if you have any other questions or feedback and suggestions on Petter's temporary solution on the **_firmware binary_**, please let us know as soon as possible. Many thanks!!!!!!

File: platform/broadcom/sonic-platform-modules-ragile/common/modules/wb_optoe.c
Comment (prgeor) [inline platform/broadcom/sonic-platform-modules-ragile/common/modules/wb_optoe.c:1]: @pettershao-ragilenetworks why not use the sonic-kernel exiting optoe driver https://github.com/sonic-net/sonic-linux-kernel/blob/e72818fcb6c2b02864e66d842d67e13d8b67df4a/patch/driver-support-optoe-QSFP_DD.patch#L75

Comment (pettershao-ragilenetworks) [reply platform/broadcom/sonic-platform-modules-ragile/common/modules/wb_optoe.c:1]: @prgeor  The reason is that: in order to fix below issue:
https://github.com/sonic-net/sonic-linux-kernel/issues/317

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@yxieca please merge

**Pull request merged**